manager/apis/app/v1alpha1/plotter_types.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PlotterSpec defines the desired state of Plotter, which is applied in a multi-clustered environment. Plotter installs the runtime environment
+// (as blueprints running on remote clusters) which provides the Data Scientist's application with secure and governed access to the data requested in the
+// M4DApplication.
+type PlotterSpec struct {
+
+	// Selector enables to connect the resource to the application
+	// Should match the selector of the owner - M4DApplication CRD.
+	Selector metav1.LabelSelector `json:"selector"`
+
+	// +required
+	// +kubebuilder::validation:MinItems:=1
+	// Blueprints structure represents remote blueprints mapped by the identifier of a cluster in which they will be running
+	Blueprints map[string]BlueprintSpec `json:"blueprints"`
```

Comment:
If the key of this map (string) is the cluster name. Is this enough to identify the blueprint? Don't we need name/namespace as well?

Reply:
For a single blueprint this is not relevant, since m4dapplication controller generates the Blueprint and reflects in its status the required metadata of the generated resource.
A Plotter is different in this respect. M4DApplication generates a Plotter resource and does not really care where the blueprints will be deployed. Plotter controller is supposed (1) to create a new namespace on a remote cluster for each blueprint and (2) to create a Blueprint in that namespace. It may use its own metadata to decide on the blueprint name, may auto-generate, as long as it keeps somewhere the information. I did not specify the full status of Plotter. It should keep a lot of data for keeping track on blueprints, including their deployment state - it is no use to track a blueprint which creation has failed, or helm releases have not been installed yet.

Reply:
Ok. I'm fine with that. Let's keep it for a later PR to update the PlotterStatus when we create a PlotterController.